windows/win_nssm.ps1

```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
 # POWERSHELL_COMMON
 
 $params = Parse-Args $args;
+
 $result = New-Object PSObject;
 Set-Attr $result "changed" $false;
 
+$name = Get-Attr $params "name" -failifempty $true
+$state = Get-Attr $params "state" "present" -validateSet "present", "absent", "started", "stopped", "restarted" -resultobj $result
 
-If ($params.name)
-{
-    $name = $params.name
-}
-Else
-{
-    Fail-Json $result "missing required argument: name"
-}
-
-If ($params.state)
-{
-    $state = $params.state.ToString().ToLower()
-    $validStates = "present", "absent", "started", "stopped", "restarted"
-
-    If ($validStates -notcontains $state)
-    {
-        Fail-Json $result "state is $state; must be one of: $validStates"
-    }
-}
-else
-{
-    $state = "present"
-}
-
-If ($params.application)
-{
-    $application = $params.application
-}
-Else
-{
-    $application = $null
-}
+$application = Get-Attr $params "application" $null
+$appParameters = Get-Attr $params "app_parameters" $null
+$startMode = Get-Attr $params "start_mode" "auto" -validateSet "auto", "manual", "disabled" -resultobj $result
```

Comment:
I would change this to:
```
$startMode = Get-Attr $params "start_mode" -validateSet "auto", "manual", "disabled" -default "auto" -resultobj $startModeResult
```
That way you are explicitly stating the result object and the default state.